src/go/rpk/pkg/cli/cmd/redpanda/admin/selftest/selftest_test.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package selftest
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClusterStatus(t *testing.T) {
+	for _, test := range []struct {
+		name            string
+		serverResponse  string
+		runningNodes    []int
+		isUninitialized bool
+	}{
+		{
+			"Test runningNodes all running case",
+			`[
+               {
+                 "node_id": 1,
+                 "status": "running"
+               },
+               {
+                 "node_id": 0,
+                 "status": "running"
+               },
+               {
+                 "node_id": 2,
+                 "status": "running"
+               }
+            ]`,
+			[]int{0, 1, 2},
+			false,
```

Comment:
Can you add the names so it's easier to read the test?

That way you can leave the zero values out (e.g: false booleans, if you want)

```
{
	name: "Test runningNodes all running case",
	serverResponse: `[
		   {
			 "node_id": 1,
			 "status": "running"
		   },
		   {
			 "node_id": 0,
			 "status": "running"
		   },
		   {
			 "node_id": 2,
			 "status": "running"
		   }
		]`,
	runningNodes:    []int{0, 1, 2},
	isUninitialized: false,
},
```

Reply:
This applies to the other tests as well.